backend/api/endpoints/team.py

```diff
@@ -25,18 +39,132 @@ def get_team_by_id(team_id: int, response: Response, session_cookie: Optional[st
 
     if user is None:
         response.status_code = 401
+        logger.info(get_team_by_id.__name__ + " /api/team/" + str(team_id) + str(response.status_code) + ": user does not exist")
         return None
 
     team_info = teamFuncs.get_team_info(team_id)
     if team_info is None:
         response.status_code = 400
+        logger.info(get_team_by_id.__name__ + " /api/team/" + str(team_id) + str(response.status_code) + ": team_info does not exist")
         return None
 
     cipher_game_id = get_game_id(team_id)
     if not is_in_team(team_id, user.person_id) and not user.is_root and not is_staff(cipher_game_id, user.person_id):
         response.status_code = 401
+        logger.info(get_team_by_id.__name__ + " /api/team/" + str(team_id) + str(response.status_code) + ": user is not in team")
         return None
 
     response.status_code = 200
     team_info.strip()
     return team_info
+
+
+@router.post('/api/team/create')
+def create_team(cipher_game_id: int, team_name: str, response: Response, session_cookie: Optional[str] = Cookie(None)) -> Optional[int]:
+
+    """
+        Creates team for given cipher game
+
+        Info can be seen by admin of game, root or player of the team
+
+        :param cipher_game_id: Id of the team to use
+        :param team_name: name of a team, that should be created
+
+        :return 200 OK if user could create team for given cipher game
+                404 NOT FOUND cipher game does not exists
+                401 UNAUTHORIZED user has no permissions to see information about team
+                409 CONFLICT team is already joined to ciphergame
+    """
+
+    user = user_management.get_user_by_token(session_cookie)
+    if user is None:
+        response.status_code = 401
+        logger.info(create_team.__name__ + " /api/createteam " + str(response.status_code) + ": user does not exist")
+        return None
+
+    given_cipher = cipherGameFuncs.get_cipher_game(cipher_game_id)
+
+    if given_cipher is None:
+        response.status_code = 404
+        logger.info(create_team.__name__ + " /api/createteam " + str(response.status_code) + ": cipher game " + str(cipher_game_id) + " does not exist")
+        return None
+
+    team = Team
+    team.name = team_name
+    team.approved = False
+
+    team_id = teamFuncs.insertTeam(team)
+
+    if team_id is None:
+        # response.status_code = ???
```

Comment:
Použij code 409, není třeba, každý kód využit jen jednou. Chybová odpověď může mít i tělíčko obsahující data o chybě, pokud je to potřeba.